### ISSUE TITLE
Database issue

### ISSUE THREAD

**Issue by XenZibe** [MillionDollarScript/MillionDollarScript#42]
Hi there, I'm having a specific issue regarding the database.
1. The Config table won't upload via the install script, It states my varchar section can't be 255, I made it 150 and it seems to be fine.
2. My ads won't display on the blocks.
3. In the database the "published" and "approved" Sections cannot be the same SET values, 

So these would normally be 'Y','N'

This cannot happen, and I reproduce this issue every time. If I change my approved to a VARCHAR and make it default to Y then the script works and notices it as approved.

In the "Publish Pixels" Section it doesn't actually show any of my orders.

Happy to give you any debug information, Using PHP 5.6.

**Comment by ty2u** [comment]
Hi there,

What versions of the MDS and MySQL are you using?
What is the exact error you get?

It seems like maybe you could be using a newer version of MySQL and I don't think I've tested it past maybe 5.5 or 5.6 for the dev snapshot. If you aren't using the dev snapshot you might wish to give it a shot.

**Comment by XenZibe** [comment]
Hi,

MySQL is running 5.6 I'm running Dev snapshot.

When I go into my approve pixels section these are the errors that I get.
Hi,

MySQL is running 5.6 I'm running Dev snapshot.

When I go into my approve pixels section these are the errors that I get.

2016/07/12 00:52:57 [error] 24267#0: *9861 FastCGI sent in stderr: "PHP message: PHP Notice:  Undefined index: BID in /home/***********/***************/www/dev/milliondollar/html/header.php on line 28
PHP message: PHP Notice:  Undefined index: BID in /home/***********/***************/www/dev/milliondollar/html/header.php on line 28" while reading response header from upstream, client: ****, server: dev.***************.co.uk, request: "GET /milliondollar/ HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "dev.***************.co.uk"
2016/07/12 00:52:57 [error] 24267#0: *9861 FastCGI sent in stderr: "PHP message: PHP Notice:  Use of undefined constant grid_width - assumed 'grid_width' in /home/***********/***************/www/dev/milliondollar/display_stats.php on line 57
PHP message: PHP Notice:  Use of undefined constant grid_height - assumed 'grid_height' in /home/***********/***************/www/dev/milliondollar/display_stats.php on line 57" while reading response header from upstream, client: ****, server: dev.***************.co.uk, request: "GET /milliondollar/display_stats.php?BID=1 HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "dev.***************.co.uk", referrer: "http://dev.***************.co.uk/milliondollar/"
2016/07/12 00:53:10 [error] 24267#0: *9868 FastCGI sent in stderr: "PHP message: PHP Notice:  Undefined index: BID in /home/***********/***************/www/dev/milliondollar/admin/approve.php on line 48
PHP message: PHP Notice:  Undefined index: action in /home/***********/***************/www/dev/milliondollar/admin/approve.php on line 59
PHP message: PHP Notice:  Undefined index: mass_approve in /home/***********/***************/www/dev/milliondollar/admin/approve.php on line 68
PHP message: PHP Notice:  Undefined index: action in /home/***********/***************/www/dev/milliondollar/admin/approve.php on line 85
PHP message: PHP Notice:  Undefined index: mass_disapprove in /home/***********/***************/www/dev/milliondollar/admin/approve.php on line 94
PHP message: PHP Notice:  Undefined index: do_it_now in /home/***********/***************/www/dev/milliondollar/admin/approve.php on line 112" while reading response header from upstream, client: ****, server: dev.***************.co.uk, request: "GET /milliondollar/admin/approve.php?app=N HTTP/1.1", upstream: "fastcgi://127.0.0.1:9000", host: "dev.*******.co.uk", referrer: "http://dev.*******.co.uk/milliondollar/admin/menu.php"
2016/07/12 00:53:10 [error] 24267#0: *9868 FastCGI sent in stderr: "PHP message: PHP Notice:  Undefined variable: order_id in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 183
PHP message: PHP Notice:  Undefined index: BID in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 186
PHP message: PHP Notice:  Undefined index: BID in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 190
PHP message: PHP Notice:  Use of undefined constant name - assumed 'name' in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 196
PHP message: PHP Notice:  Undefined index: save_links in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 205
PHP message: PHP Notice:  Undefined index: edit_links in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 225
PHP message: PHP Notice:  Undefined variable: offset in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 284
PHP message: PHP Notice:  Undefined index: offset in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 311
PHP message: PHP Notice:  Undefined index: BID in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 312
PHP message: PHP Notice:  Undefined index: do_it_now in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 322
PHP message: PHP Notice:  Use of undefined constant ID - assumed 'ID' in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 351
PHP message: PHP Notice:  Use of undefined constant banner_id - assumed 'banner_id' in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 351
PHP message: PHP Notice:  Use of undefined constant ID - assumed 'ID' in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 351
PHP message: PHP Notice:  Use of undefined constant banner_id - assumed 'banner_id' in /home/*******/*******/www/dev/milliondollar/admin/approve.php on line 351" while reading upstream, client


The **** Represent personal information that I'd rather not give out.

I dunno if this might help you, from what I'm seeing it's trying to find my stuff but it's missing lots of information in order to get it right?

**Comment by ty2u** [comment]
MDS = MillionDollarScript

I've never tested it on nginx, that could certainly be and most likely is the issue. It sure wouldn't surprise me and I suspect some sort of custom rules would have to be added for it but I'm not entirely sure which ones.

The errors you listed are actually just minor notices. I could fix them but they should get handled automatically. They are there because the original author of the script wrote some sloppy code. I've already gone through a lot of it and fixed a great deal of them but it seems you've discovered more.

I am working on a WordPress plugin version of this but it's not ready yet. It hopefully would work ok on nginx but I'm pretty sure you do have to add some custom rules for WordPress but I have a pretty good config for it I've used in the past and there should be a lot of documentation out there by now.

You could just disable reporting of notices in your error_reporting setting by setting it to E_ALL & ~E_NOTICE either in the config.php or your php.ini.

Still though it seems odd your MySQL complained it can't make a VARCHAR field 255 in length and that other error you mentioned about published and approved not being the same SET values. I'm not really sure what that means. But these were the errors I was referring to. Do you have complete error messages about those 2 errors? Where do you see errors about them? It seems like maybe there is some other weird issue going on, perhaps some MySQL config error.

**Comment by ty2u** [comment]
Hello,

I've set up a test server to see if I can test and hopefully fix this stuff better. However I used Ubuntu 16.04 so it may have more updated versions of software but this should be fine hopefully and will make sure things work in the future. I will also test on Ubuntu 14.04 afterwards to make sure it works ok there with the software it provides.

I can already see that it doesn't properly install on MySQL 5.7 so maybe you are having similar issues on MySQL 5.6.

I'm using nginx 1.9.15 and PHP 7 to test so we shall see what happens. So far, I can't even get it to install so will have to do some updating and will let you know how it goes. I may create a separate branch for this in the git.

I'm not sure how much work I want to do on this as I'd rather be working on the WordPress plugin for it which will surely be a lot nicer once I finish it.
